src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -121,7 +183,24 @@ internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
             ref byte[] output,
             uint sequenceNumber)
         {
-            throw new PlatformNotSupportedException();
+            Debug.Assert(!isNtlm, "Encrypt: NTLM is not yet supported");
+            SafeDeleteNegoContext gssContext = securityContext as SafeDeleteNegoContext;
+            byte[] tempOutput = Interop.GssApi.Encrypt(gssContext.GssContext, isConfidential, buffer, offset, count);
```

Comment:
Can securityContext be something other than a SafeDeleteNegoContext?  If no, then this should be a cast rather than using ```as```.  If yes, then we should be checking whether gssContext is null.